numpy/random/mtrand/mtrand.pyx

```diff
@@ -248,28 +248,23 @@ cdef object cont2_array(rk_state *state, rk_cont2 func, object size,
     cdef broadcast multi
 
     if size is None:
-        multi = <broadcast> PyArray_MultiIterNew(2, <void *>oa, <void *>ob)
-        array = <ndarray> PyArray_SimpleNew(multi.nd, multi.dimensions, NPY_DOUBLE)
-        array_data = <double *>PyArray_DATA(array)
-        with lock, nogil:
-            for i from 0 <= i < multi.size:
-                oa_data = <double *>PyArray_MultiIter_DATA(multi, 0)
-                ob_data = <double *>PyArray_MultiIter_DATA(multi, 1)
-                array_data[i] = func(state, oa_data[0], ob_data[0])
-                PyArray_MultiIter_NEXT(multi)
+        multi = <broadcast>PyArray_MultiIterNew(2, <void *>oa, <void *>ob)
+        array = <ndarray>PyArray_SimpleNew(multi.nd, multi.dimensions, NPY_DOUBLE)
     else:
         array = <ndarray>np.empty(size, np.float64)
-        array_data = <double *>PyArray_DATA(array)
-        multi = <broadcast>PyArray_MultiIterNew(3, <void*>array, <void *>oa, <void *>ob)
-        if (multi.size != PyArray_SIZE(array)):
+        multi = <broadcast>PyArray_MultiIterNew(3, <void *>oa, <void *>ob, <void*>array)
+        if multi.size != PyArray_SIZE(array):
             raise ValueError("size is not compatible with inputs")
-        with lock, nogil:
-            for i from 0 <= i < multi.size:
-                oa_data = <double *>PyArray_MultiIter_DATA(multi, 1)
-                ob_data = <double *>PyArray_MultiIter_DATA(multi, 2)
-                array_data[i] = func(state, oa_data[0], ob_data[0])
-                PyArray_MultiIter_NEXTi(multi, 1)
-                PyArray_MultiIter_NEXTi(multi, 2)
+
+    array_data = <double *>PyArray_DATA(array)
+
+    with lock, nogil:
+        for i from 0 <= i < multi.size:
```

Comment:
Is there some reason not to convert this ancient PyRex format loop to a modern `for i in range(multi.size)`?